### ISSUE TITLE
Process Manager doesn't handle exceptions properly in forked processes

### ISSUE THREAD

**Issue by pawel-siejba** [magento/magento2#30622]
The following bug can have fatal consequences if somebody employed any form of exception handling around indexers in their installation, as it could lead to very hard to discover bugs, that appear only on environments with several threads available, and MAGE_INDEXER_THREADS_COUNT env variable set. 
### Preconditions (*)

1. Magento 2.3.0 CE/EE or higher
2. Magento 2.4-develop
3. Several stores are active
4. (Optional) Elasticsearch is configured as search_engine - for ease of reproducing

### Steps to reproduce (*)

1. Run `MAGE_INDEXER_THREADS_COUNT=4 php bin/magento indexer:reindex catalogsearch_fulltext`
2. Kill Elasticsearch process to simulate an exception thrown

### Expected result (*)

1. Main process exits and the message is displayed only once, the exceptions from the forked processes are not handled as they were thrown in the main process.

### Actual result (*)

1. Observe multiple Indexer failure logs and messages
```
xx@xx:/var/www/magento2$ MAGE_INDEXER_THREADS_COUNT=4 php bin/magento indexer:reindex catalogsearch_fulltext
Catalog Search indexer process unknown error:
Indexer handler is not available: elasticsearch7
Catalog Search indexer process unknown error:
Indexer handler is not available: elasticsearch7
Catalog Search indexer process unknown error:
Indexer handler is not available: elasticsearch7
Catalog Search indexer process unknown error:
Indexer handler is not available: elasticsearch7
Catalog Search indexer process unknown error:
Catalog Search indexer process unknown error:
Catalog Search indexer process unknown error:
Indexer handler is not available: elasticsearch7
Indexer handler is not available: elasticsearch7
Indexer handler is not available: elasticsearch7
Catalog Search indexer process unknown error:
Fail in child process
```
---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [x] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by pawel-siejba** [comment]
@magento I am working on this

**Comment by pawel-siejba** [comment]
@magento add to contributors team

**Comment by pawel-siejba** [comment]
@magento add to contributors team

**Comment by pawel-siejba** [comment]
@magento I am working on this

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Delta 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-39193` were created

**Issue Available**: @engcom-Delta, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @pawel-siejba. Thank you for your report. 
The issue has been fixed in magento/magento2#30626 by @pawel-siejba in 2.4-develop branch
Related commit(s):
 - [5a5b6f7d4ada5ffbf55821499c547d182f8509c1](https://github.com/magento/magento2/commit/5a5b6f7d4ada5ffbf55821499c547d182f8509c1)

The fix will be available with the upcoming 2.4.3 release.
